--- a/src/wl/sys/wl_cfg80211_hybrid.c	2021-01-19 03:23:46.520331110 -0600
+++ b/src/wl/sys/wl_cfg80211_hybrid_new.c	2021-01-19 14:55:21.883479328 -0600
@@ -39,6 +39,9 @@
 #include <proto/802.11.h>
 #include <wl_cfg80211_hybrid.h>
 #include <wl_linux.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0)
+#include <linux/sched/signal.h>
+#endif
 
 #define EVENT_TYPE(e) dtoh32((e)->event_type)
 #define EVENT_FLAGS(e) dtoh16((e)->flags)

--- a/src/wl/sys/wl_linux.c	2021-01-19 03:23:46.537331311 -0600
+++ b/src/wl/sys/wl_linux_new.c	2021-01-19 14:53:45.752362108 -0600
@@ -2983,7 +2983,9 @@
 	if (skb == NULL) return;
 
 	skb->dev = wl->monitor_dev;
-	skb->dev->last_rx = jiffies;
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(4, 10, 0)
+ 	skb->dev->last_rx = jiffies;
+#endif
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 22)
 	skb_reset_mac_header(skb);
 #else
